Automattic/pocket-casts-android
# Wear App: Fix download issue

## Description
Attempt to fix: p1720414812004489/1720357905.284329-slack-C02A333D8LQ

This PR

1. Declares `foregroundServiceType` for `SystemForegroundService` for the Wear app. 

    It was added to the phone app in https://github.com/Automattic/pocket-casts-android/pull/1684 while making Android 14 changes but skipped for the Wear app. I'm declaring it for the Wear app to avoid possible issues in long-running download tasks that the system might kill.

2. Adds network security config for the Wear app similar to the phone app in an attempt to fix issues like this one:

    `java.net.SocketException: Binding socket to network 105 failed: EPERM (Operation not permitted)`

    Internal ref: p1721571942804979-slack-C02A333D8LQ

## Testing Instructions
1. Login with a Plus account to Wear app
2. Select a long-duration episode from a podcast
4. Download the episode 
5. ✅ Notice that the download completes

## Checklist
 
#### I have tested any UI changes...
